dipy/core/sphere.py

```diff
@@ -498,6 +500,155 @@ def disperse_charges(hemi, iters, const=.2):
+    weights : array-like (N, N)
+        Weight values to the electrostatic energy.
+
+    Returns
+    -------
+    energy : float
+        Sum of all interactions between any two vectors.
+
+    """
+
+    nb_points = vects.shape[0] / 3
+    # weights = kwargs.get('weights', np.ones((nb_points, nb_points)) / nb_points**2)
+    weights = kwargs.get('weights', np.ones((nb_points, nb_points)))
+    charges = vects.reshape((nb_points, 3))
+    all_charges = np.concatenate((charges, -charges))
+    all_charges = all_charges[:, None]
+    r = charges - all_charges
+    r_mag = np.sqrt((r * r).sum(-1))[:, :, None]
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        potential = 1 / r_mag ** alpha
+
+    d = np.arange(len(charges))
+    potential[d, d] = 0
+    potential = potential[:nb_points] + potential[nb_points:]
+    potential = weights * potential.sum(-1)
+    potential = potential.sum()
+    return potential
+
+
+def _get_grad_forces_alt(vects, alpha=2.0, **kwargs):
+    """1st-order derivative of electrostatic-like repulsion energy.
+
+    Parameters
+    ----------
+    vects : array-like (N * 3,)
+        Points on the sphere.
+    alpha : float
+        Controls the power of the repulsion. Default is 1.0.
+
+    Returns
+    -------
+    grad : array-like (N * 3,)
+        Gradient of the objective function.
+
+    """
+
+    nb_points = vects.shape[0] / 3
+    # weights = kwargs.get('weights', np.ones((nb_points, nb_points)) / nb_points**2)
+    weights = kwargs.get('weights', np.ones((nb_points, nb_points)))
+    charges = vects.reshape((nb_points, 3))
+    all_charges = np.concatenate((charges, -charges))
+    all_charges = all_charges[:, None]
+    r = charges - all_charges
+    r_mag = np.sqrt((r * r).sum(-1))[:, :, None]
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        forces = -2 * alpha * r / r_mag ** (alpha + 2.)
+
+    d = np.arange(len(charges))
+    forces[d, d] = 0
+    forces = forces[:nb_points] + forces[nb_points:]
+    forces = forces * weights.reshape((nb_points, nb_points, 1))
+    forces = forces.sum(0)
+    return forces.reshape((nb_points * 3))
+
+
+def disperse_charges_alt(init_pointset, iters, tol=1.0e-3):
+    """Reimplementation of disperse_charges making use of
+    scipy.optimize.fmin_slsqp.
+
+    Parameters
+    ----------
+    init_pointset : HemiSphere
+        Points on a unit sphere.
+    iters : int
+        Number of iterations to run.
+    tol : float
+        Tolerance for the optimization.
+
+    Returns
+    -------
+    array-like (N * 3,)
+        Distributed points on a unit sphere.
+
+    """
+
+    K = init_pointset.shape[0]
+    vects = optimize.fmin_slsqp(_get_forces_alt, init_pointset.reshape(K * 3),
+                                f_eqcons=_equality_constraints,
+                                fprime=_get_grad_forces_alt, iter=iters, acc=tol,
+                                args=[], iprint=0)
```

Comment:
I think in newer versions of `scipy.optimize`, `args` is a `tuple`, not a `list`. Giving a `list` as input causes a `TypeError: can only concatenate tuple (not "list") to tuple`

Reply:
Nice catch ! Yes, I checked that it is the case for the version required by DIPY as well. Thanks Emmanuel. Fixed in 059a9b8.